src/codegen.rs

```diff
@@ -1507,6 +1614,26 @@ fn codegen_expr(indent: usize, expr: &CheckedExpression, project: &Project) -> S
                     output.push_str(&codegen_expr(indent, rhs, project));
                     output.push(')');
                 }
+                BinaryOperator::Add
+                | BinaryOperator::Subtract
+                | BinaryOperator::Multiply
+                | BinaryOperator::Divide
+                | BinaryOperator::Modulo
+                    if is_integer(lhs.ty()) =>
+                {
+                    output.push_str(&codegen_checked_binary_op(indent, lhs, rhs, op, project))
+                }
+                BinaryOperator::AddAssign
+                | BinaryOperator::SubtractAssign
+                | BinaryOperator::MultiplyAssign
+                | BinaryOperator::DivideAssign
+                | BinaryOperator::ModuloAssign
+                    if is_integer(lhs.ty()) =>
```

Comment:
I think we should check the type of the expression itself instead of just the left hand side

Reply:
Done!